src/models/edge.ts

```diff
@@ -129,13 +142,16 @@ export class EdgeFactory {
     data?: Omit<IEdgeData<N, E>, 'data' | 'startNode' | 'endNode'>,
   ): IEdge<N, E> {
     const newEdge = EdgeFactory.create<N, E>({
-      data: edge.data,
+      data: edge.getData(),
       offset: data?.offset !== undefined ? data.offset : edge.offset,
       startNode: edge.startNode,
       endNode: edge.endNode,
     });
-    newEdge.state = edge.state;
-    newEdge.style = edge.style;
+    newEdge.setState(edge.getState());
+    newEdge.setStyle(edge.getStyle());
+    edge.getListeners().forEach((listener) => {
```

Comment:
It would be nice to make `getListeners()` a `private` or `protected` function, but in that case it can't be used in factory here. Should I leave it public or is there some kind of a workaround?